typescript/infra/src/agents/key-utils.ts

```diff
@@ -40,10 +47,17 @@ export function getValidatorKeys(
 ): Array<AgentKey> {
   // For each chainName, create validatorCount keys
   return agentConfig.contextChainNames.flatMap((chainName) =>
-    [...Array(agentConfig.validatorSets[chainName].validators.length)].map(
-      (_, index) =>
-        getKey(agentConfig, KEY_ROLE_ENUM.Validator, chainName, index),
-    ),
+    agentConfig.validatorSets[chainName].validators.map((validator, index) => {
```

Comment:
Noticing that this function `getValidatorKeys` is called by `getAllKeys` here https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/agents/key-utils.ts#L59, which is used in a couple spots that I think will be sorta broken if a readonly key is used:

`deleteAgentKeys`, which will throw in the `delete` function of the read only key
https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/agents/key-utils.ts#L68

`createAgentKeysIfNotExists`, which will throw in the `createIfNotExists` fn https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/agents/key-utils.ts#L79

I'd suggest we either don't throw in `createIfNotExists`, `update`, and `delete` and maybe log instead (https://github.com/abacus-network/abacus-monorepo/blob/main/typescript/infra/src/agents/agent.ts#L131), or wherever we loop through all keys to do certain things, filter by them being readonly or not